### PR TITLE
log celery task exceptions via django logger on NIC

### PR DESCRIPTION
**Pull request by sravfeyn**
@gcapalbo

### CODE REVIEW

**Comment by sravfeyn** [comment]
By default it sends task-exception emails. We use our internal loggers when we catch exceptions inside a task, but when celery task itself fails, celery uses its own logging wrapper. By default it's configured to send emails. 

As part of disabling emails on NIC, I am going to [disable](https://github.com/dimagi/commcarehq-ansible/pull/723) `CELERY_SEND_TASK_ERROR_EMAILS` on NIC and use this PR to hijack logging from  celery wrapper onto django logger. Since django logger on NIC uses sentry, these exceptions will come into sentry now.

I don't see any reason not to do this on all envs

File: corehq/util/signals.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import logging
+from celery.signals import task_failure
+from django.conf import settings
+
+
+@task_failure.connect
+def log_celery_task_exception(task_id, exception, traceback, einfo, *args, **kwargs):
+    if settings.environment == 'icds':
```
Comment (snopoke) [inline corehq/util/signals.py]: should be `settings.SERVER_ENVIRONMENT`

File: corehq/util/signals.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+        django_logger = logging.getLogger('django.request')
+        django_logger.error(u"Exception - {}".format(exception.message))
```
Comment (snopoke) [inline corehq/util/signals.py]: Can do this which will give the correct traceback etc:
```
notify_exception('Celery task failure', exec_info=einfo.exc_info)
```

**Review by snopoke — DISMISSED** [review]

**Comment by sravfeyn** [comment]
Updated

**Review by gcapalbo — APPROVED** [review]
looks good to me. just confirming that notify exception emails go to sentry?

**Comment by sravfeyn** [comment]
Yeah, they go sentry

**Pull request merged**